x-pack/plugins/apm/server/routes/alerts/rule_types/utils/get_interval_in_seconds.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { INTERVAL_MULTIPLIER_FOR_LOOKBACK } from '../../../../../common/rules/apm_rule_types';
+
+const intervalUnits = ['y', 'M', 'w', 'd', 'h', 'm', 's', 'ms'];
+const INTERVAL_STRING_RE = new RegExp(
+  '^([0-9\\.]*)\\s*(' + intervalUnits.join('|') + ')$'
+);
+
+interface UnitsToSeconds {
+  [unit: string]: number;
+}
+
+const units: UnitsToSeconds = {
+  ms: 0.001,
+  s: 1,
+  m: 60,
+  h: 3600,
+  d: 86400,
+  w: 86400 * 7,
+  M: 86400 * 30,
+  y: 86400 * 356,
+};
+
+export const getIntervalInSeconds = (interval: string): number => {
+  const matches = interval.match(INTERVAL_STRING_RE);
+  if (matches) {
+    return (
+      parseFloat(matches[1]) *
+      INTERVAL_MULTIPLIER_FOR_LOOKBACK *
+      units[matches[2]]
+    );
+  }
+  throw new Error('Invalid interval string format.');
+};
```

Comment:
This feels too low level - like something that is solved already by the platform, response ops or as a helper in AO. I'm surprised we have to do this at all - shouldn't the interval be stored as an integer, not a string?

Reply:
Yes, a function that converts "1m" to 60 is not apm specific. I'm surprised this is not a solved problem elsewhere (by an 3rd party lib that we already use, or by an internal elastic package)

Reply:
removed `getIntervalInSeconds`